llvmDebugInfoC/src/scripts/konan_lldb.py

```diff
@@ -265,16 +289,24 @@ def __none(iterable, f):
         return (i for i,v in enumerate(self._children) if v.name() == name).next()
 
     def get_child_at_index(self, index):
-        return self._values[index]
+        result = self._values[index]
+        if result is None:
+            result = self._read_value(index)
+            self._values[index] = result
+        return result
+
 
     # TODO: fix cyclic structures stringification.
     def to_string(self):
-        return dict([(self._children[i].name(), self._deref_or_obj_summary(i)) for i in range(self._children_count)])
-
+        if VISUAL_DEBUGGER:
```

Comment:
This is the primary "optimization". Visual debugger really doesn't need the summary string. Can get rid of TODO comment.